woocommerce/woocommerce-android
# Add payment canceled event for the payment flow

Closes: #5257 


### Description

This PR adds `card_present_collect_payment_cancelled` event for the payment flow. This event is triggered when the user presses the back button during the payment flow.

This event will be having a description that `User manually canceled the payment at state` along with the payment state at which the user triggered this event.
 i.e `User manually cancelled the payment at state {Loading|Collecting|Processing..etc}`

Different payment states
1. Loading
2. Collecting
3. Processing
3. Capturing
4. Success
5. Failed
6. Receipt Print

### Testing instructions

1. Navigate to the Order Detail Screen which is eligible for IPP
2. Click on the `Collect Payment` button
3. On the different payment states (listed above) press the back button
4. Check in the logs that the `card_present_collect_payment_cancelled` event is triggered with the appropriate state
5. Redo steps 2-4 on different payment states (listed above) and make sure the event is triggered with the appropriate payment state
